state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1285,7 +1290,7 @@ pub mod pallet {
 			deposit_metadata: Option<CcmDepositMetadata>,
 			tx_hash: TransactionHash,
 			deposit_details: Box<<T::TargetChain as Chain>::DepositDetails>,
-			broker_fees: Beneficiaries<T::AccountId>,
+			broker_fees: Option<(Beneficiary<T::AccountId>, Affiliates<AffiliateShortId>)>,
```

Comment:
Already discussed with @albert-llimos I believe, but I think the consensus was that we should make the broker mandatory.

Reply:
Sure, but does this also mean that we won't allow swaps into our change address? Also, I believe what @albert-llimos was discussing was mostly extenral interfaces? Can we not leave Option in this extrinsic call which is only used by our own engines? I also saw mentions of using "fallback" brokers in some edge cases and None seems like a good candidate for this.

Reply:
That's right, my idea was to require a broker Account to be witnessed as part of a Vault swap. Then we would need to check if that broker AccountId exists. If so, it's all good, and for now if it doesn't exist we would fallback to some broker or for now just let it through in some way.
However, I imagine the checking of the existence of the broker AccountId is better done in the SC, not in the engine. Therefore, it makes sense to not make this optional as we are making it mandatory in the engine witnessing. The SC will then check it's existence and fallback if necessary. So imo there is no real use of the engine passing an Option here. If in the future we make it mandatory to pass a valid AccountId, we would anyway make that change in the SC.

Reply:
Well, the use case for None right now is supporing deposits into the change address. I never got a definitive answer on whether or not we want to allow them, but I guess we are not supporting after all since the broker in mandatory?

Reply:
I don't know what the change address is :sweat_smile: